Warnings on Windows with MSYS2 + mingw-w64

**Describe the bug**
several " may be used uninitialized" and "out of bounds" warnings 

**To Reproduce**
compilation

**Expected behavior**
no warning


**Environment**
 - OS: Windows 10
 - Compiler version: g++ 11.2.0
 - CPU type: x86_64

git of today.

Here are the warnings : 

```
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/color_management.cc: In function 'jxl::Status jxl::MaybeCreateProfile(const jxl::ColorEncoding&, jxl::PaddedBytes*)':
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/color_management.cc:508:9: warning: 'void* memset(void*, int, size_t)' offset [0, 3] is out of the bounds [0, 0] [-Warray-bounds]
  508 |   memset(icc_sum.data() + 44, 0, 4);
      |   ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/color_management.cc:509:9: warning: 'void* memset(void*, int, size_t)' offset [0, 3] is out of the bounds [0, 0] [-Warray-bounds]
  509 |   memset(icc_sum.data() + 64, 0, 4);
      |   ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~

```

```
In file included from C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/chroma_from_luma.h:23,
                 from C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.h:18,
                 from C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc:6:
In member function 'jxl::ANSSymbolReader& jxl::ANSSymbolReader::operator=(jxl::ANSSymbolReader&&)',
    inlined from 'jxl::Status jxl::{anonymous}::GetBlockFromBitstream::Init(jxl::BitReader* __restrict__*, size_t, size_t, size_t, const jxl::Rect&, jxl::GroupDecCache*, jxl::PassesDecoderState*, size_t)' at C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc:613:77,
    inlined from 'jxl::Status jxl::DecodeGroup(jxl::BitReader* __restrict__*, size_t, size_t, jxl::PassesDecoderState*, jxl::GroupDecCache*, size_t, jxl::ImageBundle*, size_t, bool, bool)' at C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc:751:3:
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_ans.h:151:7: warning: '<anonymous>.jxl::ANSSymbolReader::num_special_distances_' may be used uninitialized [-Wmaybe-uninitialized]
  151 | class ANSSymbolReader {
      |       ^~~~~~~~~~~~~~~
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc: In function 'jxl::Status jxl::DecodeGroup(jxl::BitReader* __restrict__*, size_t, size_t, jxl::PassesDecoderState*, jxl::GroupDecCache*, size_t, jxl::ImageBundle*, size_t, bool, bool)':
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc:613:77: note: '<anonymous>' declared here
  613 |           ANSSymbolReader(&dec_state->code[pass + first_pass], readers[pass]);
      |                                                                             ^
In file included from C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/chroma_from_luma.h:23,
                 from C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.h:18,
                 from C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc:6:
In member function 'jxl::ANSSymbolReader& jxl::ANSSymbolReader::operator=(jxl::ANSSymbolReader&&)',
    inlined from 'jxl::Status jxl::{anonymous}::GetBlockFromBitstream::Init(jxl::BitReader* __restrict__*, size_t, size_t, size_t, const jxl::Rect&, jxl::GroupDecCache*, jxl::PassesDecoderState*, size_t)' at C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc:613:77,
    inlined from 'jxl::Status jxl::DecodeGroup(jxl::BitReader* __restrict__*, size_t, size_t, jxl::PassesDecoderState*, jxl::GroupDecCache*, size_t, jxl::ImageBundle*, size_t, bool, bool)' at C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc:751:3:
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_ans.h:151:7: warning: '*(unsigned char (*)[480])((char*)&<unnamed> + offsetof(jxl::ANSSymbolReader, jxl::ANSSymbolReader::special_distances_[0]))' may be used uninitialized [-Wmaybe-uninitialized]
  151 | class ANSSymbolReader {
      |       ^~~~~~~~~~~~~~~
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc: In function 'jxl::Status jxl::DecodeGroup(jxl::BitReader* __restrict__*, size_t, size_t, jxl::PassesDecoderState*, jxl::GroupDecCache*, size_t, jxl::ImageBundle*, size_t, bool, bool)':
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_group.cc:613:77: note: '<anonymous>' declared here
  613 |           ANSSymbolReader(&dec_state->code[pass + first_pass], readers[pass]);
      |                                                                             ^

```

```
In file included from C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/icc_codec.h:19,
                 from C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/icc_codec.cc:6:
In member function 'jxl::ANSSymbolReader& jxl::ANSSymbolReader::operator=(jxl::ANSSymbolReader&&)',
    inlined from 'jxl::Status jxl::ICCReader::_ZN3jxl9ICCReader4InitEPNS_9BitReaderEy.part.0(jxl::BitReader*, size_t)' at C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/icc_codec.cc:319:49:
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/dec_ans.h:151:7: warning: '<anonymous>.jxl::ANSSymbolReader::num_special_distances_' may be used uninitialized [-Wmaybe-uninitialized]
  151 | class ANSSymbolReader {
      |       ^~~~~~~~~~~~~~~
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/icc_codec.cc: In member function 'jxl::Status jxl::ICCReader::_ZN3jxl9ICCReader4InitEPNS_9BitReaderEy.part.0(jxl::BitReader*, size_t)':
C:/Documents/msys2/home/vincent.torri/gitroot_64/libjxl_main/lib/jxl/icc_codec.cc:319:49: note: '<anonymous>' declared here
  319 |     ans_reader_ = ANSSymbolReader(&code_, reader);
      |                                                 ^

```

FWIW, there seems to be one from pixbufloader (skcms related): https://github.com/msys2/MINGW-packages/actions/runs/3118427105/jobs/5057708500

```
  In function 'D baseline::cast(const S&) [with D = __vector(4) float; S = __vector(4) short unsigned int]',
      inlined from 'void baseline::exec_ops(const Op*, const void**, const char*, char*, int)' at C:/_/mingw-w64-libjxl/src/libjxl/third_party/skcms/src/Transform_inl.h:785:28:
  C:/_/mingw-w64-libjxl/src/libjxl/third_party/skcms/src/Transform_inl.h:114:14: warning: 'd' may be used uninitialized [-Wmaybe-uninitialized]
    114 |         d[i] = v[i];
        |         ~~~~~^~~~
  C:/_/mingw-w64-libjxl/src/libjxl/third_party/skcms/src/Transform_inl.h: In function 'void baseline::exec_ops(const Op*, const void**, const char*, char*, int)':
  C:/_/mingw-w64-libjxl/src/libjxl/third_party/skcms/src/Transform_inl.h:112:7: note: 'd' was declared here
    112 |     D d;
        |       ^
```

This is from the 0.7.0 CI build though...

> FWIW, there seems to be one from pixbufloader (skcms related): https://github.com/msys2/MINGW-packages/actions/runs/3118427105/jobs/5057708500
> 
> ```
>   In function 'D baseline::cast(const S&) [with D = __vector(4) float; S = __vector(4) short unsigned int]',
>       inlined from 'void baseline::exec_ops(const Op*, const void**, const char*, char*, int)' at C:/_/mingw-w64-libjxl/src/libjxl/third_party/skcms/src/Transform_inl.h:785:28:
>   C:/_/mingw-w64-libjxl/src/libjxl/third_party/skcms/src/Transform_inl.h:114:14: warning: 'd' may be used uninitialized [-Wmaybe-uninitialized]
>     114 |         d[i] = v[i];
>         |         ~~~~~^~~~
>   C:/_/mingw-w64-libjxl/src/libjxl/third_party/skcms/src/Transform_inl.h: In function 'void baseline::exec_ops(const Op*, const void**, const char*, char*, int)':
>   C:/_/mingw-w64-libjxl/src/libjxl/third_party/skcms/src/Transform_inl.h:112:7: note: 'd' was declared here
>     112 |     D d;
>         |       ^
> ```
> 
> This is from the 0.7.0 CI build though...
Perhaps this should be fixed in skcms, then?

--- a/lib/jxl/enc_cluster.h
+++ b/lib/jxl/enc_cluster.h
@@ -20,7 +20,10 @@
 namespace jxl {
 
 struct Histogram {
-  Histogram() { total_count_ = 0; }
+  Histogram() {
+    total_count_ = 0;
+    entropy_ = 0.0;
+  }
   void Clear() {
     data_.clear();
     total_count_ = 0;

